fastplotlib/graphics/linecollection.py

```diff
@@ -113,3 +117,35 @@ def __init__(
             )
 
             self.add_graphic(lg, reset_index=False)
+
+    def _set_feature(self, feature: str, new_data: Any, indices: Union[int, range]):
+        if not hasattr(self, "_previous_data"):
+            self._previous_data = {}
+        elif hasattr(self, "_previous_data"):
+            self._reset_feature(feature)
+        #if feature in # need a way to check if feature is in
+        feature_instance = getattr(self, feature)
+        if indices is not None:
+            previous = feature_instance[indices].copy()
```

Comment:
so for collection it's going to be a bit diff, for keeping previous data just store the new data (which becomes previous data on the next event). This will become clear in my next comments

```python
# put this import at the top of the module
from copy import deepcopy

previous = deepcopy(new_data)
```